lib/resty/core/shdict.lua

```diff
@@ -102,7 +101,7 @@ local function shdict_store(zone, op, key, value, exptime, flags)
     end
 
     if type(key) ~= "string" then
-        key = tostring(key)
+        error("bad key (string expected, got " .. type(key) .. ")")
```

Comment:
If we are to stay consistent with the ngx_lua behavior (and that of this library), we need to add the `lvl` argument to those `error()` calls, to report the proper stack level:
```lua
error("bad key (string expected, got " .. type(key) .. ")", 2)
```

Reply:
Ah I see, that's a good catch. @jdesgats Could you please update this PR to add the `lvl` argument?

Reply:
Good catch, thanks for the review! Fixed now.